Add translation context to the Extensions label in domain search filter

dlind1 opened the pull request:
#### Changes proposed in this Pull Request

Add context to 'Extensions' label in domains filter, so that it'd be possible to provide correct translations for it, instead of reusing the standard 'Extensions' which usually means 'Software Extensions'.

<img width="346" alt="Image 2021-07-01 at 3 31 21 PM" src="https://user-images.githubusercontent.com/7847633/124132894-93308900-da81-11eb-85fb-87ac3d8991d5.png">


#### Testing instructions

* Checkout the diff and `yarn start` or use Calypso.live and navigate to /start/domains
* Confirm that the page and filtering is working and the 'domains filter' button/popup displays 'Extensions:' label at the top (should be untranslated initially).

Fixes 283-gh-Automattic/i18n-issues

dlind1 committed:
Add translation context to the Extensions label in domain search filter

a8ci18n commented on client/components/domains/search-filters/dropdown-filters.jsx:
```diff
@@ -231,7 +231,7 @@ export class DropdownFilters extends Component {
 					<ValidationFieldset className="search-filters__tld-filters">
 						<FormLabel className="search-filters__label" htmlFor="search-filters-max-characters">
-							{ translate( 'Extensions' ) }:
+							{ translate( 'Extensions', { context: 'domain extension, like .com' } ) }:
```
:warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **69** times:
`translate( 'Extensions' )` *ES Score: 11*

jornp commented:
I just tested this locally, but only now saw that you didn't add the team as a reviewer yet 😆  Anyway, I can confirm that the filtering is still working, and the label is now showing "Extensions:" untranslated.

Before (Dutch, incorrect):
![image](https://user-images.githubusercontent.com/75777864/124139727-2e2c6180-da88-11eb-9c49-94d3244e4d7e.png)

After:
![image](https://user-images.githubusercontent.com/75777864/124139827-3f756e00-da88-11eb-975e-ab7e336b4c9e.png)

So this LGTM!

dlind1 commented:
> I just tested this locally, but only now saw that you didn't add the team as a reviewer yet 😆 

Thanks! I've added the team as a reviewer 😄  Also added the 'string freeze' label and would merge after it's translated.

jornp reviewed (approved):
> Also added the 'string freeze' label and would merge after it's translated.

Sounds great, LGTM!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6114962

Thank you @dlind1 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6114962

Thank you @dlind1 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6114962).
